### PR TITLE
Simplify and update evaluation code

### PR DESCRIPTION
**Pull request by atiorh**
- [x] Adopt `mlx-whisper` from PyPI and remove repo+model cloning
- [x] Adopt configs and arguments for `WhisperMLX` and `WhisperCpp` to represent apples-to-apples performance and quality.
- [x] Allow for dead simple evaluations

e.g.

```shell
pip install -e `.[evals,pipelines]` &&  whisperkit-evaluate-model --model-version openai/whisper-tiny.en --evaluation-dataset earnings22-debug --pipeline {whisper.cpp,WhisperMLX,WhisperKit} --output-dir evals
```

### CODE REVIEW

**Review by Vaibhavs10 — COMMENTED** [review]
Thanks for the update, in principle it looks okay to me, digging into the code now.

File: whisperkit/pipelines.py [hunk 1]
```diff
@@ -102,9 +103,9 @@ class WhisperKit(WhisperPipeline):
     """ Pipeline to clone, build and run the CLI from
     https://github.com/argmaxinc/WhisperKit
     """
-    _compute_unit: CoreMLSwiftComputeUnit = CoreMLSwiftComputeUnit.ANE
-    _randomize_dispatch: bool = False
     _word_timestamps: bool = False
+    _text_decoder_compute_engine = "cpuAndNeuralEngine"
+    _audio_encoder_compute_engine = "cpuAndGPU"
```
Comment (Deep-unlearning) [inline whisperkit/pipelines.py:108]: Could you explain why is the compute engine different for the encoder and the decoder ?

Comment (atiorh) [reply whisperkit/pipelines.py:108]: There is no reason,  just a default. These could be overridden with either value for either model. This makes me think I should document the intent better 👍

Comment (atiorh) [reply whisperkit/pipelines.py]: Done, also simplified the WhisperKit model downloads!

**Pull request merged**